src/ecrn_sup.erl

```diff
@@ -28,19 +28,31 @@ start_link() ->
 
 %% @private
 init([]) ->
-    RestartStrategy = one_for_one,
-    MaxRestarts = 3,
-    MaxSecondsBetweenRestarts = 10,
-    SupFlags = {RestartStrategy,
-                MaxRestarts,
-                MaxSecondsBetweenRestarts},
-
-    ChildSup =  {ecrn_cron_sup, {ecrn_cron_sup, start_link, []},
-                 permanent, 1000, supervisor, [ecrn_cron_sup]},
-    RegistrationServer  =  {ecrn_reg_server, {ecrn_reg, start_link, []},
-                            permanent, 1000, worker, [ecrn_reg]},
-    BroadcastServer  =  {ecrn_control, {ecrn_control, start_link, []},
-                         permanent, 1000, worker, [ecrn_control]},
-
-
-    {ok, {SupFlags, [ChildSup, RegistrationServer, BroadcastServer]}}.
+    SupFlags = #{
+        strategy  => one_for_one,
+        intensity => application:get_env(erlcron, sup_intensity, 3),
+        period    => application:get_env(erlcron, sup_period,   10)
+    },
+
+    ChildSup     = #{id       => ecrn_cron_sup,
+                     start    => {ecrn_cron_sup, start_link, []},
+                     restart  => permanent,
```

Comment:
The same as the previous child specs comment: These could be separate private functions whose sole purpose would be _given the arguments, create a a map containing the child specs_.

Reply:
It seems that this is a matter of taste. IMHO, this way it's a little easier to read as the supervisor specs are all named values in the map.  But feel free to change as you see fit.